src/python/pants/backend/python/tasks2/python_create_distributions.py

```diff
@@ -0,0 +1,81 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pex.interpreter import PythonInterpreter
+
+from pants.backend.python.targets.python_distribution import PythonDistribution
+from pants.backend.python.tasks2.pex_build_util import prepare_dist_workdir
+from pants.backend.python.tasks2.setup_py import SetupPyRunner
+from pants.base.exceptions import TaskError
+from pants.task.task import Task
+from pants.util.dirutil import safe_mkdir
+
+
+class PythonCreateDistributions(Task):
+  """Create python distributions (.whl) from python_dist targets."""
+
+  PYTHON_DISTS = 'user_defined_python_dists'
+
+  @classmethod
+  def product_types(cls):
+    return [cls.PYTHON_DISTS]
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    round_manager.require_data(PythonInterpreter)
+    round_manager.require_data('python')  # For codegen.
+
+  @staticmethod
+  def is_distribution(target):
+    return isinstance(target, PythonDistribution)
+
+  def __init__(self, *args, **kwargs):
+    super(PythonCreateDistributions, self).__init__(*args, **kwargs)
+
+  def execute(self):
+    dist_targets = self.context.targets(self.is_distribution)
+    built_dists = set()
+    
+    if dist_targets:
+      # Check for duplicate distribution names, since we write the pexes to <dist>/<name>.pex.
```

Comment:
this task produces wheels, not pex files tho - which I wouldn't expect to write to dist (until consumed by a pex).

afaict, its possible for two `python_dist` targets to share the same base target name - but have totally different dist names based on their setup.py files. so I'm not sure this validation buys much here, fwict? thoughts?

Reply:
We don't really know the dist_name will be created from the target definition currently.  It seems like a reasonable approximation?

Reply:
Yeah, I think you're right that this doesn't really protect against much. Just removed.